tests/chart_tests/test_nginx.py

```diff
@@ -484,3 +485,112 @@ def test_nginx_backend_overrides():
     )
 
     assert len(docs) == 0
+
+
+@pytest.mark.parametrize(
+    "kube_version",
+    supported_k8s_versions,
+)
+class TestNginxClusterRoles:
+    def test_nginx_cp_clusterrole_rbac_enabled(self, kube_version):
+        """Test that helm renders nginx control plane ClusterRole when all conditions are met."""
+        # Test control plane mode
+        values = {"global": {"rbacEnabled": True, "clusterRoles": True, "plane": {"mode": "control"}}}
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/nginx/templates/controlplane/nginx-cp-role.yaml"],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "ClusterRole"
+        assert doc["apiVersion"] == "rbac.authorization.k8s.io/v1"
+        assert doc["metadata"]["name"] == "release-name-cp-nginx"
+        assert doc["metadata"]["labels"]["tier"] == "nginx"
+        assert doc["metadata"]["labels"]["plane"] == "control"
+        assert len(doc["rules"]) > 0
+
+        # Test unified plane mode (should also work for control plane template)
+        values["global"]["plane"]["mode"] = "unified"
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/nginx/templates/controlplane/nginx-cp-role.yaml"],
+        )
+        assert len(docs) == 1
+        assert docs[0]["kind"] == "ClusterRole"
+        assert docs[0]["metadata"]["labels"]["plane"] == "unified"
+
+    def test_nginx_dp_clusterrole_rbac_enabled(self, kube_version):
+        """Test that helm renders nginx data plane ClusterRole when all conditions are met."""
+        values = {"global": {"rbacEnabled": True, "clusterRoles": True, "plane": {"mode": "data"}}}
+        docs = render_chart(
+            kube_version=kube_version,
+            values=values,
+            show_only=["charts/nginx/templates/dataplane/nginx-dp-role.yaml"],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "ClusterRole"
+        assert doc["apiVersion"] == "rbac.authorization.k8s.io/v1"
+        assert doc["metadata"]["name"] == "release-name-dp-nginx"
+        assert doc["metadata"]["labels"]["tier"] == "nginx"
+        assert len(doc["rules"]) > 0
+
+    def test_nginx_clusterrole_disabled_conditions(self, kube_version):
+        """Test that nginx ClusterRoles are not rendered when required conditions are not met."""
```

Comment:
This docstring is ambiguous. We need to know exactly what is being tested. Same for anything else that says "when required conditions are not met." We should be specific and say "When ____ is enabled and ____ is set to ______" or something.